spec/notifier_spec.rb

```diff
@@ -111,6 +111,20 @@
         dir.join("one.txt").write("hello world")
         run_thread.join
       end
+
+      it "can be stopped before it starts processing" do
+        barrier = Concurrent::Event.new
+
+        run_thread = Thread.new do
+          barrier.wait
+          @notifier.run
+        end
+
+        @notifier.stop
+        barrier.set
+
+        run_thread.join(1) or raise "timeout"
+      end
```

Comment:
This does not test the scenario where `Notifier#run` acquires the `@running` mutex before `Notifier#stop`.  I'm not sure how I would do that non-invasively.  However, it does at least replicate calling `Notifier#stop` before `Notifier#run` has started its loop.